core/projection/projection.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projection
+
+import "github.com/hazelcast/hazelcast-go-client/internal/projection"
+
+// SingleAttribute returns a projection that extracts the value of the given `attributePath`.
```

Comment:
`attributePath must not be empty`. Does this piece of comment apply here? I think it should be mentioned.